### PR TITLE
Transition from Travis CI to GitHub Actions

### PR DESCRIPTION
**Pull request by jkeys089**
## What do these changes do?

1. Bring this branch up to date with the main branch.
2. Transition from Travis CI to GitHub Actions.
3. Fix `test_version` test that was not working on python 3.8+

## Are there changes in behavior for the user?

No

## Related issue number

#137 and #196 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `235.misc`

### CODE REVIEW

**Comment by jkeys089** [comment]
Two quick notes:
1. You'll likely want to create your own [memcached action](https://github.com/jkeys089/actions-memcached). I forked [niden/actions-memcached](https://github.com/niden/actions-memcached) just to be able to run the matrix strategy without failing after the first job.
2. I'm not 100% sure the `deploy` phase is correct. Per @Dreamsorcerer I mostly just copied this from [aiohttp-session](https://github.com/aio-libs/aiohttp-session/blob/master/.github/workflows/ci.yaml)

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      uses: codecov/codecov-action@v1
+      with:
+        file: ./coverage.xml
+        flags: unit
+
+  deploy:
+    name: Deploy
+    environment: release
+    if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
+    needs: [test]
```
Comment (Dreamsorcerer) [inline .github/workflows/ci.yaml:44]: Ah, seems aiohttp-session is not as recent as I thought. Can you add the `check` step as seen in this one: https://github.com/aio-libs/aiohttp-devtools/blob/master/.github/workflows/ci.yml#L104

We'll also need the lint step added, but we can do that in another PR.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        python setup.py check -rm
+    - name: Upload coverage
+      uses: codecov/codecov-action@v1
+      with:
+        file: ./coverage.xml
+        flags: unit
+
+  deploy:
+    name: Deploy
+    environment: release
```
Comment (Dreamsorcerer) [inline .github/workflows/ci.yaml:42]: @webknjaz Can you setup the release environment please?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (jkeys089) [reply .github/workflows/ci.yaml]: Done.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (webknjaz) [reply .github/workflows/ci.yaml:52]: @asvetlov I've found another PyPI package where I'm not added as an owner.

Comment (webknjaz) [reply .github/workflows/ci.yaml:52]: @Dreamsorcerer done

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        pip install -e .
+        pip install codecov
+    - name: Run memcached service
+      uses: jkeys089/actions-memcached@master
+    - name: Run tests
+      run: |
+        make cov
+        python setup.py check -rm
+    - name: Upload coverage
+      uses: codecov/codecov-action@v1
```
Comment (webknjaz) [inline .github/workflows/ci.yaml:35]: v2?

Comment (jkeys089) [reply .github/workflows/ci.yaml]: Yes! Done.

**Comment by Dreamsorcerer** [comment]
I'll merge this so we can get started on this.

@webknjaz Any thoughts on the memcached action?

>     1. You'll likely want to create your own [memcached action](https://github.com/jkeys089/actions-memcached). I forked [niden/actions-memcached](https://github.com/niden/actions-memcached) just to be able to run the matrix strategy without failing after the first job.

**Comment by webknjaz** [comment]
@Dreamsorcerer not sure, I guess it depends on how you'd want to use that action.

**Comment by jkeys089** [comment]
> I just realised the target was py37. Could you resubmit one for master? Best to keep individual merges small. Thanks.


> @Dreamsorcerer not sure, I guess it depends on how you'd want to use that action.

@Dreamsorcerer @webknjaz I'm happy to leave it up. I just assumed you'd want to have it under your aio-libs project.

**Comment by jkeys089** [comment]
> I just realised the target was py37. Could you resubmit one for master? Best to keep individual merges small. Thanks.

@Dreamsorcerer I spent some time on this today but I wasn't able to get the current master branch to pass tests on a [version of python that still runs on ubuntu-latest](https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json). I could try to get things running on an older image / python version but I just want to be practical. Please LMK your thoughts.

**Comment by Dreamsorcerer** [comment]
OK, I'll go through it in a couple of days and maybe just use the py37 branch then.

**Pull request merged**